linked_list/linked_list.py

```diff
@@ -1,75 +1,136 @@
-        pass
+        current = self.head
+        count = 0
+        while current:
+            count += 1
+            current = current.next
+        return count
 
     # method that returns the value at a given index in the linked list
     # index count starts at 0
     # returns None if there are fewer nodes in the linked list than the index value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def get_at_index(self, index):
-        pass
+        current = self.head #initialise temp
+        count = 0 #index of current node
+        while (current):
+            if (count == index):
+                return current.value 
+            count +=1 
+            current = current.next
+        return None
 
     # method that returns the value of the last node in the linked list
     # returns None if the linked list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def get_last(self):
-        pass
+        current = self.head
+        if self.head == None: 
+            return None
+        while current.next != None:
+            current = current.next
+        return current.value
 
     # method that inserts a given value as a new last node in the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_last(self, value):
-        pass
-
+        new_node = Node(value)
+        if self.head == None:
+            self.head = new_node
+            return 
+        last = self.head
+        while (last.next):
+            last = last.next
+        last.next =  new_node
+                
     # method to return the max value in the linked list
     # returns the data value and not the node
+
     def find_max(self):
-        pass
+        head = self.head
+        if (self.head == None):
+            return None
+        else:
+            max = head.value
+            
+            while (head != None):
+                if head.value > max:
+                    max = head.value
+                head = head.next
+            return max
+            #print("Maximum value node in the list: " + str(max))
 
     # method to delete the first node found with specified value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def delete(self, value):
-        pass
+        if self.head is None:
+            return 
+
+        if self.head.value == value:
+            self.head = self.head.next
+            return
+        
+        current = self.head
+        current = self.head
+        while current.next is not None:
+            if value == current.next.value:
+                 break
+            current = current.next
+        if current.next is None:
+             print("Node not found") 
+        else:
+             current.next = current.next.next

```

Comment:
There's a couple if statements here that aren't really doing anything but stop the loop or print. It makes it hard to read what is important here. I think we can shorten it:

```suggestion
            if value == current.next.value:
                 current.next = current.next.next
            current = current.next
```
Since there could be more than one node with the same value, we don't want to `break`